compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -395,7 +391,10 @@ impl<'a> Parser<'a> {
             } else {
                 PatKind::Lit(const_expr)
             }
-        } else if self.can_be_ident_pat() {
+        // FIXME(Ezrashaw): `Parser::may_recover` is too strict here. See #109203.
```

Comment:
It's not really "too strict", it's just not really the right thing as it's about eager lookahead (which the name sadly doesn't reflect, that's my fault :D)